sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp

```diff
@@ -67,4 +53,21 @@ namespace Azure { namespace Storage {
     HashAlgorithm Algorithm;
   };
 
+  namespace Details {
+    struct CaseInsensitiveComparator
+    {
+      bool operator()(const std::string& lhs, const std::string& rhs) const
+      {
+        return std::lexicographical_compare(
+            lhs.begin(), lhs.end(), rhs.begin(), rhs.end(), [](char c1, char c2) {
+              return Core::Strings::ToLower(c1) < Core::Strings::ToLower(c2);
+            });
+      }
+    };
+
+    ContentHash FromBase64String(const std::string& base64String, HashAlgorithm algorithm);
+    std::string ToBase64String(const ContentHash& hash);
```

Comment:
This is so dirty.

Reply:
Could you please elaborate on the dirtiness? An opinion like this is almost impossible to be responded.
The problem I want to solve with those 'dirty' functions is the duplicated lines of code.

Reply:
I was whining